ERCS/erc-7579.md

```diff
@@ -268,6 +277,8 @@ interface IModule {
 }
 ```
 
+Note: A single module that is of multiple types MAY decide to pass `moduleTypeId` inside `data` to `onInstall` and/or `onUninstall` methods, so those methods are able to properly handle installation/uninstallation for various types.
```

Comment:
might be worth to give an example here as its not 100% clear to me

you're thinking smth like this right:

```solidity
// Module.sol

function onInstall(bytes memory data) external {
    (uint256 moduleTypeId, bytes memory otherData) = abi.decode(data, (uint256,bytes));
}
```

Reply:
yes.
```
if(moduleTypeId==VALIDATOR) {
   // procedure for validator
} elseif(moduleTypeId==EXECUTOR) {
  // procedure for executor
}
```